compiler/rustc_const_eval/src/transform/check_consts/qualifs.rs

```diff
@@ -170,11 +170,11 @@ impl Qualif for NeedsNonConstDrop {
             let mut selcx = SelectionContext::with_constness(&infcx, hir::Constness::Const);
             selcx.select(&obligation)
         });
-        match implsrc {
+        !matches!(
+            implsrc,
             Ok(Some(ImplSource::ConstDrop(_)))
-            | Ok(Some(ImplSource::Param(_, ty::BoundConstness::ConstIfConst))) => false,
-            _ => true,
-        }
+                | Ok(Some(ImplSource::Param(_, ty::BoundConstness::ConstIfConst)))
```

Comment:
is this a candidate for "interior patterns": `Ok(Some(ImplSource::ConstDrop(_) | ImplSource::Param(...)))`